test/alternator/test_metrics.py

```diff
@@ -103,6 +103,14 @@ def check_increases_metric(metrics, metric_names):
     for n in metric_names:
         assert saved_metrics[n] < get_metric(metrics, n, None, the_metrics), f'metric {n} did not increase'
 
+@contextmanager
+def check_increases_table_metric(metrics, metric_name, table, increase_value):
+    the_metrics = get_metrics(metrics)
+    saved_metric = get_metric(metrics, metric_name, {'cf': table}, the_metrics)
```

Comment:
Sorry, I know this is test, you use it consequently and this is nitpicking but as a newcomer I could not stop myself. It is so perfect example how you should not name identifiers:

`saved_metric = get_metric(metrics, metric_name, {'cf': table}, the_metrics)
`

Five times metrics and I have no idea which one does what :-) Feel free to ignore this comment but it is really nice candidate for refactoring.

Reply:
I've added two tests for an existing file, trying to keep the same naming convention. @nyh may have a better input

Reply:
I can understand why it's confusing. "the_metrics" is all the metrics - everything. saved_metrics are the specific metrics we are looking for (e.g., rcu_total) and it's "saved" because it's what we save before the operation.

I guess the name "the_metrics" is indeed too generic-sounding. Maybe it should have been called all_metrics or something. But Amnon is right that he just copied another similar function and use the names in it.